### ISSUE TITLE
REP

### ISSUE THREAD

**Issue by webair-studio** [ian-travers/nfsu-server#5]
![33](https://user-images.githubusercontent.com/18102692/212908906-61798f30-9b7e-499b-a675-6153d98b2ae6.jpg)
![22](https://user-images.githubusercontent.com/18102692/212908916-6f3aa229-9c4c-4214-9143-6078928a0804.jpg)
How can this be fixed? Linux Server v 2.6
Old dat files from harpyWar server. Could this be the problem?
Welcome msg types works correctly, thx.
My telegram webair_studio

p.s. Вы кажется говорите по русски, судя по комментариям в коде? :)

**Comment by webair-studio** [comment]
GetPlayerStat return NULL :(
I do not know well c++ to debug

**Comment by webair-studio** [comment]
I tried to rewrite the function, but it doesn't work.
`char* GetPlayerStat(const char* Name) {
	char tmp[1024];
	FILE* fil;

	fil = fopen("stat.dat", "rb");

	if (fil == NULL) return NULL;

	PlayerStat* it;
	
	while (!feof(fil)) {
		it = (PlayerStat*)calloc(1, sizeof(PlayerStat));
		fread(it, 1, sizeof(PlayerStat), fil);
		if (it->Name[0] != 0) {
			if (strcmp(it->Name, Name) == 0) {
				sprintf(tmp, "%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x,%x",
				it->Rating_All, it->Wins_All, it->Loses_All, it->Disc_All, it->REP_All, it->OppsREP_All, it->OppsRating_All,
				it->Rating_Circ, it->Wins_Circ, it->Loses_Circ, it->Disc_Circ, it->REP_Circ, it->OppsREP_Circ, it->OppsRating_Circ,
				it->Rating_Sprint, it->Wins_Sprint, it->Loses_Sprint, it->Disc_Sprint, it->REP_Sprint, it->OppsREP_Sprint, it->OppsRating_Sprint,
				it->Rating_Drag, it->Wins_Drag, it->Loses_Drag, it->Disc_Drag, it->REP_Drag, it->OppsREP_Drag, it->OppsRating_Drag,
				it->Rating_Drift, it->Wins_Drift, it->Loses_Drift, it->Disc_Drift, it->REP_Drift, it->OppsREP_Drift, it->OppsRating_Drift);
				fclose(fil);
				return tmp;
			}
		}
		
	}
	
	fclose(fil);
};`

**Comment by ian-travers** [comment]
@webair-studio 

> Old dat files from harpyWar server. Could this be the problem?

Yes it could! Server gets a persona name from rusers.dat. V 2.6 format was changed. Each persona is 16 bytes length instead 100 bytes.
